example/app/AppComponents.scala

```diff
@@ -0,0 +1,32 @@
+import com.gu.googleauth.{AuthAction, GoogleAuthConfig}
+import controllers.{Application, Login, routes}
+import play.api.libs.ws.ahc.AhcWSComponents
+import play.api.mvc.AnyContent
+import play.api.routing.Router
+import play.api.{ApplicationLoader, BuiltInComponentsFromContext}
+import play.filters.HttpFiltersComponents
+import router.Routes
+
+class AppComponents(context: ApplicationLoader.Context)
+  extends BuiltInComponentsFromContext(context)
+  with AhcWSComponents
+  with HttpFiltersComponents{
+
+  val clientId: String = configuration.get[String]("your.clientId.config.path")
+  val clientSecret: String = configuration.get[String]("your.clientSecret.config.path")
+  val redirectUrl: String = configuration.get[String]("your.redirectUrl.config.path")
+  val domain: String = configuration.get[String]("your.apps-domain.config.path")
+  val googleAuthConfig = GoogleAuthConfig(clientId, clientSecret, redirectUrl, domain)
+
+  val authAction = new AuthAction[AnyContent](googleAuthConfig, routes.Login.loginAction())(
```

Comment:
This was actually based on the Play documentation. https://www.playframework.com/documentation/2.6.x/ScalaActionsComposition is where I found an example of using custom actions and that follows the model of having it as a constructor parameter with the associated lowercase convention.